dynamodb/src/main/scala/akka/stream/alpakka/dynamodb/impl/DynamoSettings.scala

```diff
@@ -5,20 +5,59 @@
 package akka.stream.alpakka.dynamodb.impl
 
 import akka.actor.ActorSystem
+import com.amazonaws.auth._
+import com.typesafe.config.Config
+
+import scala.util.Try
 
 object DynamoSettings {
-  def apply(system: ActorSystem): DynamoSettings = {
-    val config = system.settings.config.getConfig("akka.stream.alpakka.dynamodb")
+
+  /**
+   * Scala API: Creates [[DynamoSettings]] from the [[com.typesafe.config.Config Config]] attached to an [[akka.actor.ActorSystem]].
+   */
+  def apply(system: ActorSystem): DynamoSettings = apply(system.settings.config)
+
+  /**
+   * Scala API: Creates [[DynamoSettings]] from a [[com.typesafe.config.Config Config]]
+   */
+  def apply(baseConfig: Config): DynamoSettings = {
+    val config = baseConfig.getConfig("akka.stream.alpakka.dynamodb")
```

Comment:
Please pass in the resolved config instance instead so that the user may choose the location in the settings to allow for several configurations.

Reply:
Passed in config resolved from `akka.stream.alpakka.dynamodb`